pkg/crypto/tinkcrypto/primitive/composite/ecdh1pu/ecdh1pu_aes_aead_private_key_manager.go

```diff
@@ -0,0 +1,192 @@
+	}
+
+	curve, err := validateKeyFormat(keyFormat.Params)
+	if err != nil {
+		return nil, errInvalidECDH1PUAESPrivateKeyFormat
+	}
+
+	keyFormat.Params.KwParams.KeyType = commonpb.KeyType_EC
+
+	pvt, err := hybrid.GenerateECDHKeyPair(curve)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ecdh1pupb.Ecdh1PuAeadPrivateKey{
+		Version:  ecdh1puAESPrivateKeyVersion,
+		KeyValue: pvt.D.Bytes(),
+		PublicKey: &ecdh1pupb.Ecdh1PuAeadPublicKey{
+			Version: ecdh1puAESPrivateKeyVersion,
+			Params:  keyFormat.Params,
+			X:       pvt.PublicKey.Point.X.Bytes(),
+			Y:       pvt.PublicKey.Point.Y.Bytes(),
+		},
+	}, nil
+}
+
+// NewKeyData creates a new KeyData according to the specification of ECDHESPrivateKey Format.
+// It should be used solely by the key management API.
+func (km *ecdh1puAESPrivateKeyManager) NewKeyData(serializedKeyFormat []byte) (*tinkpb.KeyData, error) {
+	key, err := km.NewKey(serializedKeyFormat)
+	if err != nil {
+		return nil, err
+	}
+
+	serializedKey, err := proto.Marshal(key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &tinkpb.KeyData{
+		TypeUrl:         ecdh1puAESPrivateKeyTypeURL,
+		Value:           serializedKey,
+		KeyMaterialType: tinkpb.KeyData_ASYMMETRIC_PRIVATE,
+	}, nil
+}
+
+// PublicKeyData returns the enclosed public key data of serializedPrivKey
+func (km *ecdh1puAESPrivateKeyManager) PublicKeyData(serializedPrivKey []byte) (*tinkpb.KeyData, error) {
+	privKey := new(ecdh1pupb.Ecdh1PuAeadPrivateKey)
+
+	err := proto.Unmarshal(serializedPrivKey, privKey)
+	if err != nil {
+		return nil, errInvalidECDH1PUAESPrivateKey
+	}
+
+	serializedPubKey, err := proto.Marshal(privKey.PublicKey)
+	if err != nil {
+		return nil, errInvalidECDH1PUAESPrivateKey
+	}
+
+	return &tinkpb.KeyData{
+		TypeUrl:         ecdh1puAESPublicKeyTypeURL,
+		Value:           serializedPubKey,
+		KeyMaterialType: tinkpb.KeyData_ASYMMETRIC_PUBLIC,
+	}, nil
+}
+
+// DoesSupport indicates if this key manager supports the given key type.
+func (km *ecdh1puAESPrivateKeyManager) DoesSupport(typeURL string) bool {
+	return typeURL == ecdh1puAESPrivateKeyTypeURL
+}
+
+// TypeURL returns the key type of keys managed by this key manager.
+func (km *ecdh1puAESPrivateKeyManager) TypeURL() string {
+	return ecdh1puAESPrivateKeyTypeURL
+}
+
+// validateKey validates the given ECDH1PUPrivateKey and returns the KW curve.
+func (km *ecdh1puAESPrivateKeyManager) validateKey(key *ecdh1pupb.Ecdh1PuAeadPrivateKey) (elliptic.Curve, error) {
+	err := keyset.ValidateKeyVersion(key.Version, ecdh1puAESPrivateKeyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ecdh1pu_private_key_manager: invalid key: %s", err)
+	}
+
+	return validateKeyFormat(key.PublicKey.Params)
+}
+
+// validateKeyFormat validates the given ECDHESKeyFormat and returns the KW Curve.
+func validateKeyFormat(params *ecdh1pupb.Ecdh1PuAeadParams) (elliptic.Curve, error) {
+	c, err := hybrid.GetCurve(params.KwParams.CurveType.String())
+	if err != nil {
+		return nil, err
```

Comment:
I would wrap this error with a message. It is applicable to every construction such as:
```
if err != nil {
		return nil, err
}
```